### PR TITLE
Fix #61. Prepare for publication the speakers section. Optimize the script for the modal window.

### LINKED ISSUE #61: Add Keynote Speakers section
We will be going to announce our first Keynote Speaker soon. Speaker section needs to be uncommented in `_layouts/default.html` and made ready for Keynote speaker announcement.

### PR DESCRIPTION
**Pull request by glebkema**
Fix #61.

hydpyconf2019-speakers-section

### Speakers section

The block with `<hr>` moved from `_layouts\default.html` to `_includes\speakers-section.html`, because they work in tandem with each other.

I removed the `target="_blank"` form the "Get Tickets" link because this block is located on the same page.

The block `_includes\schedule-section.html` becomes visible only when `_data\speakers.yml` exists and is not empty. To prevent this block from being published right now, I commented out all the lines in `_data\speakers.yml`.

```jekyll
{%assign speaker_code = speaker[0] %}
{%assign speaker_data = speaker[1] %}
```
I added these auxiliary variables to remember that when you go through mapping, the key and value have indexes 0 and 1 responsively.

### New way to get data for a modal window

Since there will be many references of speakers on the page, it is not very cool to encumber each modal link with all the necessary data. 

Because of this I left only one attribute `data-speaker` with the code of the desired speaker (the same as in the schedule of workshops). I take the rest of the information from the speaker data array, which I added to the list of JS scripts.

### Social networks without limits

Accounts in social networks are collected in a subitem `social`. Thanks to this, arrays with the field names that were in the previous version of the script version became unnecessary. Now the script just goes through all the fields in sequence.
 
Social links are set in pairs: "**keyword for the Font Awesome icon**" - "**URl of the social account**". The script will process links to any social networks whose icons are in the Font Awesome. It will put them in the order in which the links are listed in the speaker data.

### CODE REVIEW

**Comment by ananyo2012** [comment]
Changes look good. Can you uncomment a speaker data for now so that I can see the preview then comment it once it's ready to merge?

Also the objective of the issue was to display only Keynote speakers in the section. Please add a Heading titled "Keynote Speakers" at the beginning of the section. What you can do is add a `keynote` field in the data to check if the speaker is a keynote or not and display only the keynote speakers.

**Comment by glebkema** [comment]
How should the keynote speakers be distinguished from all others?
Add an extra property to their data?

**Comment by ananyo2012** [comment]
@glebkema Yup that's what I suggested.

**Comment by glebkema** [comment]
@ananyo2012 Done:
1. Header is "Keynote Speakers".
2. The `keynote` field. Show only the speakers with `keynote: true`.
3. Setup test data for the sections of speakers and shedule. Only one speaker is keynoted.

File: _includes/speakers-section.html
Comment (ananyo2012) [inline _includes/speakers-section.html:9]: Please move this line before assigning the variables. You can simple check using `speaker.keynote`

File: _includes/speakers-section.html
Comment (ananyo2012) [inline _includes/speakers-section.html]: Remove the tickets button here. We only want it in the hero block.

File: assets/js/main.js
Comment (ananyo2012) [inline assets/js/main.js:98]: Nice :+1:  thanks for figuring this out. I was also observing this issue in the last PR you submitted. Though I observed it only once.

**Review by ananyo2012 — APPROVED** [review]
Some minor comments else LGTM

**Comment by ananyo2012** [comment]
Also comment out the data now. Will merge this once you make the updates.

File: _includes/speakers-section.html
Comment (glebkema) [reply _includes/speakers-section.html:9]: I tried this form of recording a couple of times, but, unfortunately, the data stops getting on the page.

Perhaps the reason is that instead of listing in the data file I use mapping. Perhaps the fore loop processes such data somehow differently. Or I just don’t know something :)

File: _includes/speakers-section.html
Comment (glebkema) [reply _includes/speakers-section.html]: Done

**Comment by glebkema** [comment]
> Also comment out the data now

Done

File: _includes/speakers-section.html
Comment (ananyo2012) [reply _includes/speakers-section.html:9]: The format of yml file in `speakers.yml` seems to be incorrect. You need to put a `-` at the starting of each block of data. Can you add that and update this ?

Comment (glebkema) [reply _includes/speakers-section.html:9]: It is not a list format.  This file uses the mapping format instead. This is neccessary to access a specific speaker. Just like in Jekyll documentation:
https://jekyllrb.com/docs/datafiles/#example-accessing-a-specific-author

Comment (bhansa) [reply _includes/speakers-section.html:9]: This should be a good tool to test yml file: http://www.yamllint.com/

Comment (glebkema) [reply _includes/speakers-section.html:9]: Looks like a valid one.

yaml-check

Comment (ananyo2012) [reply _includes/speakers-section.html:9]: @glebkema Got that. So yaml list and hash fomat are 2 different things. Yaml list have an array like structure so each object is packed within an array element, while hashes have key, value pairs represented with an array of 2 elements. Here each data object is within a value.

**Review by ananyo2012 — APPROVED** [review]

File: _includes/speakers-section.html
Comment (glebkema) [reply _includes/speakers-section.html:9]: @ananyo2012  Thanks for explanation. I have not yet fully understood these terms, and your resume is very helpful.

**Comment by ananyo2012** [comment]
@glebkema @bhansa I wrote a blog post as a note to understand the differences between YAML lists and dictionaries in Jekyll https://ilovefoss.wordpress.com/2019/11/02/jekyll-yaml-lists-vs-dictionaries/

**Comment by glebkema** [comment]
@bhansa @ananyo2012 
Thanks! Now it has become convenient to test knowledge using examples from the article. How data is organized and how to access it in each case.

**Pull request merged**
